lib/gpu_memory_service/client/memory_manager.py

```
@@ -605,6 +637,230 @@ def reallocate_all_handles(self, tag: str = "default") -> None:
+        # Scratch is process-local and always needs RW access for torch.zeros
+        # and kernel writes — independent of any GMS-granted lock mode.
+        try:
+            for offset in range(0, aligned_size, self.granularity):
+                cumem_map(va + offset, self.granularity, scratch_handle)
+            cumem_set_access(va, aligned_size, self.device, GrantedLockType.RW)
+        except Exception:
+            # On partial failure, best-effort teardown and re-raise.
+            try:
+                cumem_unmap(va, aligned_size)
+            except Exception:
+                pass
```

Ruff (0.15.11)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)